src/main.js

```diff
@@ -228,10 +235,56 @@ function viewSavedPostersPage() {
   // All posters in savedPosters array is shown in the saved posters grid section
   for (var i = 0; i < savedPosters.length; i++) {
     savedPosterGrid.innerHTML +=
-    `<article class="mini-poster">
+    `<article class="mini-poster" id=${i}>
         <img class="poster-img" src=${savedPosters[i].imageURL} alt="nothin' to see here">
         <h2 class="poster-title">${savedPosters[i].title}</h2>
         <h4 class="poster-quote">${savedPosters[i].quote}</h4>
       </article>`
   }
+  addEventListenerForMiniPosters();
 };
+
+function addEventListenerForMiniPosters() {
+  var allMiniPosters = document.querySelectorAll(".mini-poster");
```

Comment:
I agree we may need to refactor or figure out another solution here though the code does function.